server/channels/app/plugin.go

```diff
@@ -253,13 +253,13 @@ func (ch *Channels) initPlugins(c *request.Context, pluginDir, webappPluginDir s
 		ch.srv.Log().Error("Failed to sync plugins from the file store", mlog.Err(err))
 	}
 
-	plugins := ch.processPrepackagedPlugins(prepackagedPluginsDir)
-	pluginsEnvironment = ch.GetPluginsEnvironment()
-	if pluginsEnvironment == nil {
-		ch.srv.Log().Info("Plugins environment not found, server is likely shutting down")
-		return
+	if err := ch.processPrepackagedPlugins(prepackagedPluginsDir); err != nil {
+		ch.srv.Log().Error("Failed to process prepackaged plugins", mlog.Err(err))
 	}
-	pluginsEnvironment.SetPrepackagedPlugins(plugins)
+	ch.pluginClusterLeaderListenerID = ch.srv.AddClusterLeaderChangedListener(func() {
+		ch.persistTransitionallyPrepackagedPlugins()
+	})
+	ch.persistTransitionallyPrepackagedPlugins()
```

Comment:
Calling is a no-op if not the cluster leader, but we want to try right away in case we are.